### PR TITLE
Awesome Books (ES6 version) 

### PR DESCRIPTION
**Pull request by katarighe**
This pull request includes the Awesome Book ES6 project to use the latest version of ES6 functionalities.

In this section, the following activities have been done: 

1. **Cloned** the [old repo](https://github.com/katarighe/awesome-books) to the new one for as `awesome-books-es6`.
2. **Set up** the project using git, linters and other various tools relating to it. 
3. **Updated** the JavaScript syntax to the new ES6 version. 
4. **Added** new features to the project, such as support for classes and modules.
5. **Fixed** any bugs that that were identified in the previous version of the project.
6. **Divided** the `index.js` file into two different JS files which were created inside the modules. 
7. **Refactored** the functions to arrow `=>` functions on ES6. 
8. **Displayed** the current date below the navigation bar by using the Luxon library.

### CODE REVIEW

File: modules/thisbook.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable */
```
Comment (frempongdev) [inline modules/thisbook.js]: It is ok to sometimes disable some linters on some lines of code to create an allowance for your application to work. However, it is not a good practice to disable linters for a whole file when you have about 2 or 3 errors to fix. ♻️ 

- [x] Kndly check and fix the errors being thrown by your linter checker as the main purpose for installing linters is to help your learning and make you are better developer. 👍🏿

File: modules/thisbook.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return `
+        <div class= "booklist">
+        <p>"${book.title}" by "${book.author}"</p>
+        <button onClick="thisBook.removeBook(${books.indexOf(book)})">Remove</button>
+        </div>
+        `;
+      })
+      .join('');
+    return bookListHTML;
+  }
```
Comment (frempongdev) [inline modules/thisbook.js]: Good job separating your code into `Modules` as expected in the requirements. 📁 However another requirement of this milestone is to refactor all your JavaScript functions and make them ES6 functions. 

- [x] Please check and use arrow functions in all the functions as it is the recommended approach for modern development. 🚀

File: modules/thisbook.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  addNewBook(bookTitle, bookAuthor) {
+    const storedBooks = this.getStoredBooks();
+    const newBook = {
+      title: bookTitle,
+      author: bookAuthor,
+    };
+    storedBooks.push(newBook);
+    this.updateStoredBooks(storedBooks);
+    this.displayBooks();
+  }
```
Comment (frempongdev) [inline modules/thisbook.js]: - [ ] Kindly check and fix the function to add books as it is not working as expected. 👍🏿 

### Please check application behavior below ⬇️ 
cannot add book

File: index.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    <header>
+        <nav class="nav-bar">
+                <a class="brand" href="./index.html">Awesome Books</a>
+                <ul class="nav-menu">
+                   <li class="nav-item"><a class="nav-link booklist-2" href="#list">Booklist</a></li>
+                    <li class="nav-item"><a class="nav-link form-section" href="#form">Add a Book</a></li>
+                    <li class="nav-item"><a class="nav-link contact-section" href="#contact">Contact</a></li>
+                </ul>
+        </nav>
+    </header>
```
Comment (frempongdev) [inline index.html]: Great implementation of dynamic display of elements, including the add book page. 🔥 However there seem to be a bug that comes along with the implementation. 

- [x] Kindly make sure that the `Add Book` navigation link in your `Navbar` does not also disappear when you are hiding the book form. 

### Please check below ⬇️ 
missing add book

File: index.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            </ul>
+        </section>
+    </div>
+
+    <!-- Footer -->
+    <footer class="footer">
+        <div class="footer-container">
+            <p class="copyright">©2023 All rights reserved.</p>
+        </div>
+    </footer>
```
Comment (frempongdev) [inline index.html:63]: You have done a good job using a footer in this milestone. The footer however does not seem to stay as the foot of the screen. 😢 

- [ ] Kindly consider using `CSS position properties` like `fixed` or absolute to keep it at the bottom of the screen. ⬇️ 

image

**Review by frempongdev — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🚧 

Hi @katarighe   👋🏼 
I'm Frempong.

You've done a great job implementing most of the requirements for this project 🚀, but there are a few changes I recommend be made. You're almost done, keep up the good work! 🤖 🎆 



 ## Highlights
✅ Great use of GitHub flow. 💯 
✅ No Linter errors. 🆗  
✅ Your project is documented professionally. 📜 
✅ Great implementation of Modules. ✔️ 


 ## Required Changes ♻️
_Kindly Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @katarighe, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 

## Optional suggestions
- Please note that to make your repo more professional, it is advised that your branch name should be descriptive, it could describe a feature or action you are working on. In this case, for example: `refactor_using_es6`. Please note this 👍 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**